### PR TITLE
build: Generate copyright headers automatically

### PR DESCRIPTION
**Pull request by apparentlymart**
HashiCorp legal now requires a copyright claim in a comment at the top of every substantial file in this repository. If we don't add this ourselves then a bot will open a PR to add missing entries, but that process adds git history, pull request, and GitHub notification noise so instead we'll deal with it proactively as part of our usual code generation steps.

This means that pull requests will fail their checks if there are any files that lack copyright headers, so we can deal with those before we merge rather than in a subsequent PR.

Adding this new external tool dependency has implicitly upgraded some of our existing dependencies, so I recommend that we wait until after cutting the v1.5 release branch and then merge this to `main` once it's representing v1.6 so that we'll have plenty of time to let these dependency upgrades soak before v1.6.0 final.

### CODE REVIEW

**Comment by jbardin** [comment]
This seems to include a whole lot of dependency upgrades as well. Are those intentional?

**Review by alisdair — APPROVED** [review]
This is a neat way to do this. I'm surprised there are so many dependency updates.

**Comment by apparentlymart** [comment]
The number of dependency updates here is, I think, a consequence of the fact that the "copywrite" tool is built in quite a similar way to Terraform itself -- relying on many of the same libraries -- but it was built more recently and so `go get` has defaulted to newer versions of those libraries in that tool's `go.mod` file. This is the "ratchet effect" of the Go modules minimum version selection design.

It did make me a little nervous -- hence the suggestion to wait until we could merge this into the v1.6 branch -- but I think with enough room for these changes to soak it's better to upgrade these rather than continuing to let them languish at older versions.

I'm going to merge this once I've resolved the inevitable merge conflicts that have emerged in the meantime.

**Pull request merged**
